server/src/test/java/org/elasticsearch/cluster/NodeConnectionsServiceTests.java

```diff
@@ -292,8 +286,8 @@ public void testOnlyBlocksOnConnectionsToNewNodes() throws Exception {
             future6.actionGet(); // completed even though the connection attempt is still blocked
             assertConnectedExactlyToNodes(nodes1);
 
-            nodeConnectionBlocks.clear();
```

Comment:
I think we risk the same problem on line 262?

Reply:
Yes, you're quite right. It needs a slightly more subtle delay to reproduce. Addressed in 
4052b5f.